src/powershell/Public/ConvertTo-FinOpsSchema.ps1

```diff
@@ -0,0 +1,148 @@
+    Converts the input data found in EA_ActualCost.csv, exporting all available columns, and stores the converted data in EA_ActualCost_Output.csv. 
+
+
+    .LINK
+    https://aka.ms/ftk/ConvertTo-FinOpsSchema
+#>
+
+param(
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ActualCost,
+
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Destination,
+
+    [Parameter(Mandatory = $false)]
+    [bool]
+    $ExportAllColumns = $true
+)
+
+function Get-AccountType {
+    [CmdletBinding()]
+    param(
+        [PSCustomObject]
+        $Object
+    )
+
+    $ftk_AccountType = switch ($true) {
+        { $Object.BillingAccountId -eq $Object.BillingProfileId } { "EA" }
+        { $Object.BillingAccountId.Contains(":") } { "MCA" }
+        default { "Other" }
+    }
+    
+    return $ftk_AccountType
+}
+
+
+function ConvertTo-FinOpsSchema {
+    [CmdletBinding()]
+    param(
+        [string]
+        $ActualCost,
+
+        [string]
+        $Destination,
+
+        [bool]
+        $ExportAllColumns
+    )
+
+    if (-not (Test-Path -Path $ActualCost)) {
+        Write-Error "Input file $ActualCost does not exist."
+        return
+    }
+    
+    $outputDirectory = [System.IO.Path]::GetDirectoryName($Destination)
+    if (-not (Test-Path -Path $outputDirectory -PathType Container)) {
+        New-Item -Path $outputDirectory -ItemType Directory
+    }
+
+    $csvData = Import-Csv -Path $ActualCost
+    
+    $rowCount = $csvData.Count
+    $processedCount = 0
+    $start = [DateTime]::Now
+    
+    $transformedData = $csvData | ForEach-Object {
+        $object = $_
+    
+        # Create a new object with the mapped column names
+        # This will ensure that the output CSV has the correct column names
+        $newObject = @{
+            BillingAccountId     = $object.BillingAccountId
+            BillingAccountName   = $object.BillingAccountName
+            ftk_AccountType      = (Get-AccountType -Object $object)
+            BillingProfileId     = $object.BillingProfileId
+            SubscriptionId       = $object.SubscriptionId
+            #... other mappings ...
+        }
+
+    # TODO: 
+        # Unexpected Columns: When an unexpected column is encountered, the name of the column is added to $unexpectedColumns to notify the user after processing.
+        # $RemoveCustomColumns: When $RemoveCustomColumns is $false (or not set), the unexpected columns are allowed to pass through to the output data. 
+        # If $RemoveCustomColumns is $true, unexpected columns are omitted from the output. Write-Warning: A single warning with all unexpected column names is issued after processing, avoiding repetitive warnings during processing.
+            
+        $processedCount++
+        
+        # Time Estimation Logic.
+        # If we have processed less than 10 rows, we will use a constant value of .01 seconds per row.
```

Comment:
Can you save this to a hard-coded variable aligned with this comment? Also, do you recall the original estimate you came up with? I forgot, but I want to make sure whatever we use is based on reality. We can always update it later.

Reply:
Yes, tested it by including estimate processing code inside my script. With the estimate processing function I created earlier the processing time for 29,719 rows with 24 columns. Difference between the script with Write-Progress: Total processing time: 4.8115966 seconds and the Estimate-Processing function I created earlier.
Estimated total processing time: 5,35 seconds
Total processing time: 6.2117018 seconds
Estimated total processing time: 7,04 seconds 
Total processing time: 4.8581067 seconds
Estimated total processing time: 5,44 seconds 
I ran it a few times. The first is this script now and the second is the function I had written earlier. The results do change a bit when you run it several times of course. Overall it is a difference of roughly 1 second.

Reply:
What do you mean by this? Can you save this to a hard-coded variable aligned with this comment?

Reply:
Not much... This is it. I also suggested a change to the reference below.
```suggestion
        # If we have processed less than 10 rows, we will use an estimated seconds per row based on testing.
        $estimatedSecPerRow = 0.01

```

Reply:
done